Missionframework/modules/06_arsenal/fnc/fn_arsenal_checkGear.sqf

```diff
@@ -37,11 +36,12 @@ if (((backpack player) != "") && ((backpack player) != _backpack)) then {_player
 {if (_x != "") then {_playerItems pushBackUnique _x;}} forEach (secondaryWeaponItems player);
 {if (_x != "") then {_playerItems pushBackUnique _x;}} forEach (handgunItems player);
 
-if (({(_x in KPLIB_arsenal_whitelist) || ((_x find "ACRE") != -1) || ((_x find "tf_") != -1) || ((_x find "TFAR_") != -1)} count _playerItems) != count _playerItems) then {
-    private _text = format ["[KP LIBERATION] [BLACKLIST] Found %1 at Player %2", (_playerItems - KPLIB_arsenal_whitelist), name player];
+// Search for forbidden items
+private _badItems = _playerItems arrayIntersect KPLIB_preset_arsenal_blacklist;
+
+if (!(_badItems isEqualTo [])) then {
```

Comment:
You could change this to `exitWith`.
Then you can get rid of the above `_return` variable and provide at scripts ends just `true` and at the end of the exitWith `false`

Reply:
Nice idea, I'll think about it